tool-openssl/rehash.cc

```diff
@@ -406,12 +404,14 @@ bool RehashTool(const args_list_t &args) {
   regex_t regex;
   int ret = regcomp(&regex, "^[0-9a-fA-F]{8}\\.([r])?[0-9]+$", REG_EXTENDED | REG_NOSUB);
   if (ret) {
+    regfree(&regex);
```

Comment:
NP:
* `regex` should be initialized to NULL in case the call to `regcomp` fails and it doesn't modify `regex`.
* Is it ok to call `regfree` with a NULL value? Documentation doesn't say: https://linux.die.net/man/3/regfree